### PR TITLE
An alternate voting method

### PR DESCRIPTION
**Pull request by CrazyAmphibian**
[controversial] [vote]

## What this does
adds a new voting method which reduces the probability of the current item being voted for. (eg, if it is a map, the currently-loaded map will have a reduced chance of winning than it normally would)
the exact numbers is that it reduces voting power by 25%, though this number can be changed easily in a var. This reduction is done last in the checks, so an option will never be dropped due to not getting enough votes because of this change.

## Why it's good
more map variety leads to a wider variety of gameplay and reduces burnout of players. Additionally, having more attention on other maps will lead to more issues in them being found and (theoretically) fixed.

## How it was tested
it wasn't. i just couldn't get my local server to compile the maps.

## Changelog
:cl:
 * tweak: map voting will now use a slightly changed voting method to promote more map variety

### CODE REVIEW

**Comment by hacker-on-steroids** [comment]
persistent voting revival when

**Comment by boy2mantwicethefam** [comment]
More often I see the number of players decreasing because of unpopular maps being picked by the RNG, not Box being voted for 3 times in a row.

**Comment by CrazyAmphibian** [comment]
> More often I see the number of players decreasing because of unpopular maps being picked by the RNG, not Box being voted for 3 times in a row.

i think this is really hard data to actually get given that
A) rounds typically are around 2 hours long
B) the server's population is only really active for 7 hours in a day
C) one map is played 50-60% of the time (which dilutes the data)

**Comment by SECBATON-GRIFFON** [comment]
i wish i couldve gotten instant runoff to work, ill try it again someday

**Comment by falcon2346** [comment]
Box babies in shambles. If this actually ends up having a noticeable effect on the pop it can always be reverted

**Comment by SonixApache** [comment]
this again?

**Comment by CrazyAmphibian** [comment]
> i wish i couldve gotten instant runoff to work, ill try it again someday

what you could do instead is allow players to vote for all maps instead of just one to have a similar effect

**Comment by toomykins** [comment]
you cannot force us to play horizon

**Comment by CrazyAmphibian** [comment]
\>you cannot force us to play horizon
isn't horizon going to be removed from the pool?

File: code/modules/html_interface/voting/voting.dm [hunk 1]
```diff
@@ -177,6 +185,40 @@ var/global/datum/controller/vote/vote = new()
...
+			if(filteredchoices[a] / total_votes < vote_threshold)
+				discarded_votes += filteredchoices[a]
+				filteredchoices -= a
+				discarded_choices += a
+			if (a==previous_pick) //the penalty is applied after the threshold to ensure it isn't dropped arbitrarily
+				filteredchoices[a]*=previous_penalty
+		if(filteredchoices.len)
+			winner = pickweight(filteredchoices.Copy())
+		qualified_votes = total_votes - discarded_votes
+		text += "<b>Random Weighted Vote Result: [winner] won with [tally[winner]] vote\s and a [round(100*tally[winner]/qualified_votes)]% chance of winning.</b>"
```
Comment (west3436) [inline code/modules/html_interface/voting/voting.dm:214]: Might risk a div/0 here - ensure qualified_votes can't ever be 0 or null.

**Review by west3436 — CHANGES_REQUESTED** [review]
This will require a server vote if you want to implement it as the primary voting method.

File: code/modules/html_interface/voting/voting.dm (hunk 1 above)
Comment (CrazyAmphibian) [reply code/modules/html_interface/voting/voting.dm:214]: 0 votes is already accounted for earlier in the code i'm pretty sure.
plus this section is copy+paste from weighted voting

Comment (CrazyAmphibian) [reply code/modules/html_interface/voting/voting.dm:214]: yeah, line 200
```
		if (total_votes > 0)
			[...]
		else
		text += "<b>Vote Result: Inconclusive - No Votes!</b>"
	return text

**Comment by CrazyAmphibian** [comment]
> This will require a server vote if you want to implement it as the primary voting method.

this is why i self-labled with [vote]

**Review by west3436 — APPROVED** [review]

**Comment by bathosbathtime** [comment]
Server poll made for this PR: https://ss13.moe/index.php/poll/263

**Comment by Lumbermancer** [comment]
I just don't know. Test merge for a week.

**Comment by wotisjanitor** [comment]
Hello adacovsk

**Comment by Killette2** [comment]
If you make it alive on the shuttle 3 votes
If you make it out on a pod or get stranded 2 votes
If you die 1 vote
This will solve all problems and I will not elaborate

**Pull request closed without merging**